### PR TITLE
Fix core data warnings when logging with different user

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/13

### 🎯 Goal
Fix the CoreData warnings when logging out and logging in with different user.

### 🛠 Implementation
After some research I found that the issue was that we were creating multiple NSManagedObjectModel's, and we need to make sure we just have one. 

Example:
```
022-11-17 17:27:03.014799+0000 ChatSample[12745:354336] [error] warning: Multiple NSEntityDescriptions claim the NSManagedObject subclass 'ChannelListQueryDTO' so +entity is unable to disambiguate.
CoreData: warning: Multiple NSEntityDescriptions claim the NSManagedObject subclass 'ChannelListQueryDTO' so +entity is unable to disambiguate.
2022-11-17 17:27:03.014894+0000 ChatSample[12745:354336] [error] warning:  	 'ChannelListQueryDTO' (0x60000026b180) from NSManagedObjectModel (0x60000162f520) claims 'ChannelListQueryDTO'.
CoreData: warning:  	 'ChannelListQueryDTO' (0x60000026b180) from NSManagedObjectModel (0x60000162f520) claims 'ChannelListQueryDTO'.
2022-11-17 17:27:03.014928+0000 ChatSample[12745:354336] [error] warning:  	 'ChannelListQueryDTO' (0x6000002742c0) from NSManagedObjectModel (0x60000169eb70) claims 'ChannelListQueryDTO'.
CoreData: warning:  	 'ChannelListQueryDTO' (0x6000002742c0) from NSManagedObjectModel (0x60000169eb70) claims 'ChannelListQueryDTO'.
2022-11-17 17:27:03.016002+0000 ChatSample[12745:354336] [error] warning: Multiple NSEntityDescriptions claim the NSManagedObject subclass 'ChannelReadDTO' so +entity is unable to disambiguate.
CoreData: warning: Multiple NSEntityDescriptions claim the NSManagedObject subclass 'ChannelReadDTO' so +entity is unable to disambiguate.
2022-11-17 17:27:03.016347+0000 ChatSample[12745:354336] [error] warning:  	 'ChannelReadDTO' (0x60000026bd30) from NSManagedObjectModel (0x60000162f520) claims 'ChannelReadDTO'.
CoreData: warning:  
```

### 🧪 Manual Testing Notes
- Open Demo App with Debugger in Xcode
- Logout
- Login with different user
- No warning was reported in the Console

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Database/DatabaseContainer.swift [hunk 1]
```diff
@@ -74,6 +74,8 @@ class DatabaseContainer: NSPersistentContainer {
     private let deletedMessageVisibility: ChatClientConfig.DeletedMessageVisibility?
     private let shouldShowShadowedMessages: Bool?
 
+    static var cachedModel: NSManagedObjectModel?
```
Comment (martinmitrevski) [inline Sources/StreamChat/Database/DatabaseContainer.swift:77]: This is never set to nil? Trying to understand what would be the impact of that.

File: Sources/StreamChat/Database/DatabaseContainer.swift [hunk 2]
```diff
@@ -99,16 +101,23 @@ class DatabaseContainer: NSPersistentContainer {
...
+        if let cachedModel = Self.cachedModel {
+            managedObjectModel = cachedModel
+        } else {
+            // It's safe to unwrap the following values because this is not settable by users and it's always a programmer error.
+            let bundle = bundle ?? Bundle(for: DatabaseContainer.self)
+            let modelURL = bundle.url(forResource: modelName, withExtension: "momd")!
+            let model = NSManagedObjectModel(contentsOf: modelURL)!
+            managedObjectModel = model
+        }
+        Self.cachedModel = managedObjectModel
```
Comment (martinmitrevski) [inline Sources/StreamChat/Database/DatabaseContainer.swift:114]: this line should go in the `else`.

**Review by martinmitrevski — COMMENTED** [review]
LGTM, just few questions

File: Sources/StreamChat/Database/DatabaseContainer.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChat/Database/DatabaseContainer.swift:77]: Yes this not set to `nil`, only initially is `nil`. We need to understand if we created it once or not

File: Sources/StreamChat/Database/DatabaseContainer.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChat/Database/DatabaseContainer.swift:114]: Right now, changing this does not impact the logic. Do you think it is more readable, tho?

Comment (nuno-vieira) [reply Sources/StreamChat/Database/DatabaseContainer.swift]: Changed it so that the logic is easier to understand 👍

**Review by martinmitrevski — APPROVED** [review]
✅

**Pull request merged**